app/models/concerns/content_matching_operations.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module ContentMatchingOperations
+  extend ActiveSupport::Concern
+
+  included do
+    def content_match?(value)
+      if pdf?
+        pdf_match?(value)
+      elsif xml?
+        content.match?(value)
+      end
+    end
+
+    def pdf_match?(value)
+      io = StringIO.new
+      io.set_encoding Encoding::BINARY
+      io.write content
+      last_page_text = ""
+      PDF::Reader.open(io) do |pdf|
+        pdf.pages.each do |page|
+          if (last_page_text + " " + page.text).gsub("\n", " ").match?(value)
+            io.close
+            return true
+          end
+
+          last_page_text = page.text
```

Comment:
Tejto logike uplne nerozumiem. jednak toto bude nachylne na whitespace a diakritiku a dvak preco matchujeme len predchadzajucu page a tu nasledujucu? Co ak je text cez 3 strany? (protipriklad)

Reply:
Tiez sa mi to velmi nepaci, ale skor teda daj pls nejaky tip, ako na to. Kedze som nic lepsie nevymyslel

Co sa tyka 3 stran, tak to by som povazoval za taky extrem, ze to asi neriesime. Zatialco rozseknutie hladaneho textu na 2 strany asi moze celkom bezne nastat.

Reply:
Ja by som to asi cele hodil do jedneho stringu a matchoval tam.